src/Xamarin.Android.Build.Tasks/Tasks/CreateMultiDexMainDexClassList.cs

```diff
@@ -32,13 +35,21 @@ public override bool Execute ()
 			Log.LogDebugTaskItems ("  CustomMainDexListFiles:", CustomMainDexListFiles);
 			Log.LogDebugMessage ("  ToolExe: {0}", ToolExe);
 			Log.LogDebugMessage ("  ToolPath: {0}", ToolPath);
-			
+			Log.LogDebugMessage ("  ProguardHome: {0}", ProguardHome);
+
 			if (CustomMainDexListFiles != null && CustomMainDexListFiles.Any ()) {
 				var content = string.Concat (CustomMainDexListFiles.Select (i => File.ReadAllText (i.ItemSpec)));
 				File.WriteAllText (MultiDexMainDexListFile, content);
 				return true;
 			}
 
+			// Windows seems to need special care.
+			// On the other hand, xbuild has a bug and fails to parse '=' in the value, so we skip JAVA_TOOL_OPTIONS on Mono runtime.
+			EnvironmentVariables =
+				OS.IsWindows ?
```

Comment:
There are two properties which are only set in msbuild: `$(MSBuildRuntimeType)` and `$(MSBuildAssemblyVersion)`. The former has values - `Full`, `Core` and `Mono`. If the properties have any value, then it should indicate that the host is msbuild. I think you might want to use the former though.